### PR TITLE
ww autocompletion: first draft

### PR DESCRIPTION
**Pull request by TeXitoi**
Add a http://localhost:5000/v1/places?q=8%20mart%20ber%20par you need an elasticsearch server with data inside to test.

### CODE REVIEW

File: source/jormungandr/jormungandr/interfaces/v1/Places.py [hunk 1]
```diff
@@ -51,11 +55,78 @@
...
+            "name": source["name"],
+            "administrative_region": marshal_es_admin(source)
+        }
+    return place
+
+class marshal_es:
+    def __call__(self, f):
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            r = f(*args, **kwargs)
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/interfaces/v1/Places.py]: I think it's better to destructure the response like:

```
response, status_code = f(*args, **kwargs)
```

File: source/jormungandr/jormungandr/interfaces/v1/Places.py [hunk 2]
```diff
@@ -86,19 +164,107 @@ def __init__(self, *args, **kwargs):
...
+                            ]
+                        }
+                    }
+                }
+            }
+        }
+        try:
+            # TODO: get params?
+            es = Elasticsearch()
+            res = es.search(index="munin", size=args['count'], body=query)
```
Comment (l-vincent-l) [inline source/jormungandr/jormungandr/interfaces/v1/Places.py:265]: I don't understand where is the configuration.
How do we know which ES is called?

Comment (TeXitoi) [inline source/jormungandr/jormungandr/interfaces/v1/Places.py:265]: By default, it'll use localhost:9200. The TODO is about being able to specify hosts in configuration.

File: source/jormungandr/jormungandr/interfaces/v1/Places.py [hunk 3]
```diff
@@ -86,19 +164,107 @@ def __init__(self, *args, **kwargs):
...
-        response = i_manager.dispatch(args, "places",
-                                      instance_name=self.region)
+
+        if not any([region, lon, lat]):
+            return self.get_ww(args)
+        else:
+            return self.get_region(args, region, lon, lat)
+
+    @marshal_with(places)
+    def get_region(self, args, region=None, lon=None, lat=None):
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/interfaces/v1/Places.py]: I don't think this method is correctly named (and there is another get_region method which get the region :p)

File: source/jormungandr/jormungandr/interfaces/v1/Places.py [hunk 4]
```diff
@@ -86,19 +164,107 @@ def __init__(self, *args, **kwargs):
...
-        g.use_old_disruptions = args['_use_old_disruptions']
         self._register_interpreted_parameters(args)
-
+        g.use_old_disruptions = args['_use_old_disruptions']
         if len(args['q']) == 0:
             abort(400, message="Search word absent")
-        response = i_manager.dispatch(args, "places",
-                                      instance_name=self.region)
+
+        if not any([region, lon, lat]):
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/interfaces/v1/Places.py]: maybe put a comment here

**Comment by antoine-de** [comment]
you should add authentication test (because it's handled manually in 'places' now)

and maybe one access test on the global place API (and check the 'no elastic search available' error)

**Pull request merged**